src/natcap/invest/utils.py

```diff
@@ -58,28 +113,6 @@ def capture_gdal_logging():
     Returns:
         ``None``
     """
-    osgeo_logger = logging.getLogger('osgeo')
-
-    def _log_gdal_errors(err_level, err_no, err_msg):
-        """Log error messages to osgeo.
-
-        All error messages are logged with reasonable ``logging`` levels based
-        on the GDAL error level.
-
-        Args:
-            err_level (int): The GDAL error level (e.g. ``gdal.CE_Failure``)
-            err_no (int): The GDAL error number.  For a full listing of error
-                codes, see: http://www.gdal.org/cpl__error_8h.html
-            err_msg (string): The error string.
-
-        Returns:
-            ``None``
-        """
-        osgeo_logger.log(
-            level=GDAL_ERROR_LEVELS[err_level],
-            msg='[errno {err}] {msg}'.format(
-                err=err_no, msg=err_msg.replace('\n', ' ')))
-
     gdal.PushErrorHandler(_log_gdal_errors)
     try:
```

Comment:
Ah!  Every `try/except` block has an optional `finally` clause:

```python
try:
    something_that_might_break()
except:
    logger.exception('it broke!')
finally:
    do_needed_cleanup()
```

In our case, the `finally` lets us ensure that no matter what happens, whether it worked great or _even if an exception was raised_, we want to pop the error handler we pushed. 

In practice, `try` only needs to be accompanied by either an `except` or a `finally`.  So we sometimes do this in the tests:

```
try:
    raster = gdal.OpenEx(raster_path)
    band = raster.GetRasterBand(1)
    array = band.ReadAsArray()
    numpy.testing.assert_allclose(array, expected_array)
finally:
    band = None
    raster = None
```
That way if something happened and the raster or band can't be opened, we can still close the datasets so that the test fixtures can remove the files as expected.